Library/Homebrew/cmd/info.rb

```diff
@@ -29,6 +29,13 @@ def print_info
         puts unless i == 0
         begin
           info_formula Formulary.factory(f)
+        rescue TapFormulaAmbiguityError
+          rack = HOMEBREW_CELLAR/f
+          if rack.directory?
+            info_formula Formulary.factory(rack.to_s)
+          else
```

Comment:
It looks like most of the places where we'd need to rescue the exception are written or can be rewritten to follow the pattern:

```rb
begin
  formula = Formulary.factory(some_variable)
rescue TapFormulaAmbiguityError
  rack = HOMEBREW_CELLAR/some_variable
  if rack.directory?
    formula = Formulary.factory(rack.to_s)
  else
    raise
  end
end
```

That indicates to me that this logic should be handled inside the call to factory. That would avoid the need to rescue this exception in most cases and simplify the logic in others.

I tentatively take back what I said about moving the rack logic out of factory.